src/Logger-Arduino_AllPins/main.py

```diff
@@ -26,61 +26,103 @@
 # SOFTWARE.
 
 
-
 # SweepMe! device class
 # Type: Logger
 # Device: Arduino AllPins
 
-import numpy as np
-import time
 
 from EmptyDeviceClass import EmptyDevice
 
+
 class Device(EmptyDevice):
+    description = """
+        <h3>Arduino AllPins</h3>
+        <p>This driver allows to read out given digital and analog pins.</p>
+        <p>&nbsp;</p>
+    """
 
     def __init__(self):
         EmptyDevice.__init__(self)
-        
+
         self.shortname = "Arduino AllPins"
-        
+
         self.variables = []
         self.units = []
-        
-        for i in np.arange(12)+2:
-            self.variables.append("Digital %i" % i)
-            self.units.append("")
-            
-        for i in np.arange(8):
-            self.variables.append("Analog %i" % i)
-            self.units.append("")
-
-        self.plottype = [True for x in self.variables] # True to plot data
+
+        self.plottype = [True for x in self.variables]  # True to plot data
         self.savetype = [True for x in self.variables]  # True to save data
-        
-        #self.idlevalue = 0 # motor position to go home
-                 
-        self.port_manager = True  
+
+        self.port_manager = True
         self.port_types = ["COM"]
-        self.port_properties = { "timeout":3,
-                                 "EOL": "\n",
-                                 "baudrate":115200,
-                                }
+        self.port_properties = {
+            "timeout": 3,
+            "EOL": "\n",
+            "baudrate": 115200,
+        }
+
+        self.max_voltage = 5
+
+        self.unit_dict = {
+            "Volt": "V",
+            "Bit": " ",
+        }
+
+    def set_GUIparameter(self):
+        return {
+            "Digital channel": "2,3,4,5,6,7,8,9,10,11,12,13",
+            "Analog channel": "0,1,2,3,4,5,6,7",
+            "Analog unit": ["Volt", "Bit"],
```

Comment:
this could be "V" instead of "Volt"

Reply:
ok, I see now that you make a translation to "V" later, so it is fine.
Maybe "Analog return type" instead of "Analog unit" with options "Voltage" and "Bit level" ?

Reply:
one could also remove or replace all "A" in the string in case people prefer to use "A0, A1, ...." which would be then supported as well.